docs/Tutorials/Permissioning/Upgrade-Permissioning-Contract.md

```diff
@@ -53,58 +48,125 @@ yarn run build
+    ```
+
+1. Update environment variables for contracts that are to be deployed. For example:
+
+    ```bash
+    RETAIN_ADMIN_CONTRACT=true
+    RETAIN_NODE_RULES_CONTRACT=false
+    RETAIN_ACCOUNT_RULES_CONTRACT=false
+    ```  
+
+### 5. Deploy the contracts
+
+1. In the `permissioning-smart-contracts` directory, deploy the contracts:
+
+    ```bash
+    truffle migrate --reset
+    ```
+
+    !!! important
+
+        - If updating from v1 to v2, the new `NodeIngress` contract address must be specified when restarting the Besu nodes. Copy this address from the migration output. For example:
+        ```bash
+        > Deployed NodeIngress contract to address = 0x12B1f953395080A13AeED0dC4d0bb14e787A91cF
+        ```
+        - If upgrading from v2 (using separate storage contracts) copy the `Storage` contract addresses displayed in the output. For example:
+        ```
+        > Deployed NodeStorage contract to address = 0x4F3e35A3Be3C1b77Ade39969D175C743ad3484Ee
+        ...
+        > Deployed AccountStorage contract to address = 0x2362187023D738034B516438Af187356b31E8Fb8
+        ```
+
+1. Set the storage contract address environment variables to ensure that the storage contracts are not re-deployed. For example:
+
+    ```bash
+    NODE_STORAGE_CONTRACT_ADDRESS=0xE0bF6021e023a197DBb3fABE64efA880E13D3f4b
+    ACCOUNT_STORAGE_CONTRACT_ADDRESS=0x7153CCD1a20Bbb2f6dc89c1024de368326EC6b4F
+    ```
+
+1. Deploy the updated contracts:
+
+    ```bash
+    truffle migrate --reset
+    ```
+
+### 6. Start the Permissioning Management Dapp
+
+In the `permissioning-smart-contracts` directory, start the Web server:
+
+```bash
+yarn start
+```
+
+The Dapp displays at [`http://localhost:3000`](http://localhost:3000).
+
+### 7. Restart Besu nodes
+
+Restart the Besu nodes with the updated [`NodeIngress`](#4-deploy-the-contract)
+contract address and [permissioning contract interface](../../HowTo/Limit-Access/Specify-Perm-Version.md)
 version.
 
 ```cmd
-besu --data-path=data --genesis-file=../cliqueGenesis.json --permissions-accounts-contract-enabled --permissions-accounts-contract-address "0x3d49d1eF2adE060a33c6E6Aa213513A7EE9a6241" --permissions-nodes-contract-enabled  --permissions-nodes-contract-address "0x4E72770760c011647D4873f60A3CF6cDeA896CD8" --permissions-nodes-contract-version=2 --rpc-http-enabled --rpc-http-cors-origins="*" --rpc-http-api=ADMIN,ETH,NET,PERM,CLIQUE --host-allowlist="*"
+besu --data-path=data --genesis-file=../cliqueGenesis.json --permissions-accounts-contract-enabled --permissions-accounts-contract-address "0x0000000000000000000000000000000000008888" --permissions-nodes-contract-enabled  --permissions-nodes-contract-address "0x4E72770760c011647D4873f60A3CF6cDeA896CD8" --permissions-nodes-contract-version=2 --rpc-http-enabled --rpc-http-cors-origins="*" --rpc-http-api=ADMIN,ETH,NET,PERM,CLIQUE --host-allowlist="*"
 ```
 
-In the command, the following options have been updated:
+In the command, note the following options:
 
```

Comment:
It seems this and the following bullets points can be deleted.

Reply:
It might be repetition but I would be inclined to keep these points esp the "permissions-nodes-contract-version"

Reply:
changed my mind! I added "version 2" on the line before the command instead.